webview-ui/src/components/chat/chat-view/components/layout/WelcomeSection.tsx

```diff
@@ -23,10 +24,12 @@ export const WelcomeSection: React.FC<WelcomeSectionProps> = ({
 	taskHistory,
 	shouldShowQuickWins,
 }) => {
-	const { lastDismissedInfoBannerVersion, lastDismissedCliBannerVersion, lastDismissedModelBannerVersion } = useExtensionState()
+	const { lastDismissedInfoBannerVersion, lastDismissedCliBannerVersion, lastDismissedModelBannerVersion, activeBanners } =
+		useExtensionState()
 
 	const shouldShowInfoBanner = lastDismissedInfoBannerVersion < CURRENT_INFO_BANNER_VERSION
 	const shouldShowNewModelBanner = lastDismissedModelBannerVersion < CURRENT_MODEL_BANNER_VERSION
+	const topBanners = (activeBanners || []).filter((banner) => banner.placement === "top")
```

Comment:
Might worth memorize this should that it gets updated when changes and won't get re-render on each mount?

```suggestion
	const topBanners = useMemo(() => activeBanners?.filter(banner => banner.placement === "top", [activeBanners])
```

Then below you can do:

```
{topBanners?.map((banner) => (
					<ApiBanner banner={banner} key={banner.id} />
				))}
```